pkg/logging/cri_logger.go

```diff
@@ -126,13 +140,48 @@ func ReadLogs(opts *LogViewOptions, stdout, stderr io.Writer, stopChannel chan o
 					return fmt.Errorf("failed to read log file %q: %v", logPath, err)
 				}
 				if opts.Follow {
-
+					if !continueNext {
+						return nil
+					}
 					// Reset seek so that if this is an incomplete line,
 					// it will be read again.
 					if _, err := f.Seek(-int64(len(l)), io.SeekCurrent); err != nil {
 						return fmt.Errorf("failed to reset seek in log file %q: %v", logPath, err)
 					}
 
+					if watcher == nil {
+						// Initialize the watcher if it has not been initialized yet.
+						if watcher, err = fsnotify.NewWatcher(); err != nil {
+							return fmt.Errorf("failed to create fsnotify watcher: %v", err)
+						}
+						defer watcher.Close()
+						if err := watcher.Add(dir); err != nil {
+							return fmt.Errorf("failed to watch directory %q: %w", dir, err)
+						}
+						// If we just created the watcher, try again to read as we might have missed
+						// the event.
+						continue
+					}
+
+					var recreated bool
+					// Wait until the next log change.
+					continueNext, recreated, err = waitLogs(context.Background(), opts.ContainerID, baseName, watcher)
+					if err != nil {
+						return err
+					}
+					if recreated {
+						newF, err := openFileShareDelete(logPath)
+						if err != nil {
+							if os.IsNotExist(err) {
```

Comment:
the file is recreate, so it should absolutely exist otherwise we need to raise error